src/main/java/com/conveyal/datatools/manager/jobs/FeedUpdater.java

```diff
@@ -264,18 +322,75 @@ private void updatePublishedFeedVersion(String feedId, FeedVersion publishedVers
      * could be that more than one versions were recently "published" and the latest published version was a bad
      * feed that failed processing by RTD.
      */
-    private static FeedVersion getLatestVersionSentForPublishing(String feedId, FeedSource feedSource) {
+    static Map<String, FeedVersion> getLatestVersionsSentForPublishing(Collection<FeedSource> feedSources) {
+        /* Corresponding mongoshell query:
+        db.getCollection('FeedVersion').aggregate([
+        {
+            $match: {
+                sentToExternalPublisher: { $exists: 1 },
+                //feedSourceId: {$in: <array>}
+            }
+        },
+        {
+            $group: {
+                _id: "$feedSourceId",
+                latestSentToExternalPublisher: { $max: "$sentToExternalPublisher" },
+                items: { $push: "$$ROOT" }
+            }
+        },
+        {
+            $unwind: "$items"
+        },
+        {
+            $match: {
+                $expr: { $eq: ["$items.sentToExternalPublisher", "$latestSentToExternalPublisher"] }
+            }
+        },
+        {
+            "$replaceRoot": {
+                "newRoot": "$items"
+            }
+        }
+        ])
+        */
+
+        List<String> feedSourceIds = feedSources.stream().map(fs -> fs.id).collect(Collectors.toList());
+
+        List<Bson> stages = Lists.newArrayList(
```

Comment:
Is there potential for this to fail silently if there is a change to the underlying fields? Could this be guarded against perhaps by checking for the existence of all required fields first?

Reply:
The check for the only required field `sentToExternalPublisher` besides `_id` is already being done in the `match` clause.

Reply:
If I change `feedSourceId` or `latestSentToExternalPublisher` to "TEST", Mongo does not throw an SQL exception. Perhaps the test `shouldGetLatestVersionsSentToPublisher` failing is enough to cover this if fields are changed in the future? I understand that this is an edge case, but an RDMS would throw an exception for missing/unknown field.

Reply:
Correct, this has to do with how Mongo fundamentally works. If the field is renamed for some reason, the publishing feature will silently fail, but also the test `shouldGetLatestVersionsSentToPublisher` that ties the Java field with the Mongo field will fail too as you mention, so we should be covered.